libraries/AP_Compass/AP_Compass.h

```diff
@@ -109,7 +109,7 @@ friend class AP_Compass_Backend;
     /// @returns heading in radians
     ///
     float calculate_heading(const Matrix3f &dcm_matrix) const {
-        return calculate_heading(dcm_matrix, 0);
+        return calculate_heading(dcm_matrix, get_first_usable());
```

Comment:
should have a private variable uint8_t first_usable, which is updated once per update call

Reply:
done, please review. I have also retested by flying SITL copter.